lib/teleterm/gateway/gateway.go

```diff
@@ -92,12 +92,20 @@ func New(cfg Config) (*Gateway, error) {
 	cfg.LocalPort = port
 
 	gateway := &Gateway{
-		Config:       cfg,
-		closeContext: closeContext,
-		closeCancel:  closeCancel,
-		localProxy:   localProxy,
+		Config:             cfg,
+		closeContext:       closeContext,
+		closeCancel:        closeCancel,
+		localProxy:         localProxy,
+		cliCommandProvider: cliCommandProvider,
 	}
 
+	cliCommand, err := cliCommandProvider.GetCommand(gateway)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	gateway.CLICommand = cliCommand
```

Comment:
We add the `cliCommandProvider` but also add the `cliCommand` _provided_
Can we just add the provider itself?

:nail_care: 
If we need both, I think we should set it when intializing (so move this 6 lines above the initialization and just set the `CLICommand` as part of the initialization)

Reply:
Ah, I see what you mean, instead of having a field for both the provider and the command itself, let's just have the provider and then a method which returns the command, right?

Yeah that seems reasonable, idk why I didn't think of it. I swear I feel half of my programming experience goes out of the window when I'm writing in a new lang. 😅 But I hope I'll get over that soon!

Reply:
Done in 7bd935f.